dotCMS/src/test/java/com/dotmarketing/filters/CMSUrlUtilTest.java

```diff
@@ -0,0 +1,31 @@
+package com.dotmarketing.filters;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.UnsupportedEncodingException;
+import javax.servlet.http.HttpServletRequest;
+import org.junit.Test;
+
+/**
+ * @author nollymar 
+ */
+public class CMSUrlUtilTest {
+
+    /**
+     * Method To Test: {@link CMSUrlUtil#getURIFromRequest(HttpServletRequest)}
+     * Given Scenario: Call {@link CMSUrlUtil#getURIFromRequest(HttpServletRequest)}
+     * with a request that contains a mocked URI with a plus sign
+     * ExpectedResult: The URI returned must be the same one mocked,
+     * including the plus sign
+     */
+    @Test
+    public void testGetURIFromRequest() throws UnsupportedEncodingException {
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        when(request.getRequestURI()).thenReturn("dotcms+test.txt");
+        String result = CMSUrlUtil.getInstance().getURIFromRequest(request);
+        assertEquals("dotcms+test.txt", result);
+    }
+
```

Comment:
I know is out from the issue scope but I think is good idea create a test setting the CMS_FILTER_URI_OVERRIDE request attribute

Reply:
Done